### ISSUE TITLE
WebViewExt.loadFinishedEvent not working

### ISSUE THREAD

**Issue by slaven3kopic** [Notalib/nativescript-webview-ext#15]
### Which platform(s) does your issue occur on?
- iOS
- 12.1
- emulator and device

### Please, provide the following version numbers that your issue occurs with:

- CLI: 5.0.3
- Cross-platform modules: 5.0.5
- Runtime(s): 5.0.0
- Plugin(s): 5.0.0-alpha.2 and 0.5.1

WebViewExt.loadFinishedEvent callback function never executed.

`
    webViewLoaded(args: LoadEventData) {
        if (!this.webViewInterface) {
            console.log('webview loaded');
            this.handleLoader(true);
            this.webViewInterface = args.object;
            args.object.src = ~/ATR_Map/www/dist/map.html' ;
            this.webViewInterface.on(WebViewExt.loadFinishedEvent, (args2: LoadFinishedEventData) => {
                console.log('webview load finished');
                this.loadData();
            });
            this.webViewInterface.on(WebViewExt.shouldOverrideUrlLoadingEvent,
                (args2: ShouldOverideUrlLoadEventData) => {
                console.log('webview override');
                args2.cancel = true;
            });
        }
    }
`

`
    <GridLayout width="100%" height="100%" rows="*" columns="*" class="">
        <WebViewExt row="0" column="0" class="web-view" #mapWebView id="mapWebView" (loaded)="webViewLoaded($event)" debugMode="true"></WebViewExt>
    </GridLayout>

`

**Log:**

CONSOLE LOG file:///app/tns_modules/atr-nativescript-auth/auth.service.js:51:36: navigating to /map
CONSOLE LOG file:///app/ATR_Map/map-layout.component.js:23:24: webview loaded
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:160:29: NOTA: WebViewExt.src = "~/ATR_Map/www/dist/map.html" startsWith ~/ resolved to "file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html"
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:160:29: NOTA: WebViewExt.src = "~/ATR_Map/www/dist/map.html" escaped to "file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html"
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:160:29: NOTA: WKWebViewWrapper.loadUrl("file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html") -> this.ios.loadFileURLAllowingReadAccessToURL("file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html", "file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html"
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:160:29: NOTA: WebViewExt.src = "~/ATR_Map/www/dist/map.html" - LoadUrl("file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html")
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:160:29: NOTA: webViewDecidePolicyForNavigationActionDecisionHandler: "file:///Users/slaven/Library/Developer/CoreSimulator/Devices/EB5EB69C-F0F7-4872-A00A-69223DC47E92/data/Containers/Bundle/Application/54848858-4891-4DC1-B3A1-E56980165833/mobile.app/app/ATR_Map/www/dist/map.html"
CONSOLE LOG file:///app/ATR_Map/map-layout.component.js:32:28: webview override

**Comment by m-abs** [comment]
Hi @slaven3kopic 

I think this is caused by your `shouldOverrideUrlLoadingEvent`.

`args2.cancel = true` tells the `WebView` to not load the url, the page is never loaded and `loadFinishedEvent` shouldn't be triggered.

**Comment by nbvishwa** [comment]
@slaven3kopic can i get to know how to use the plugin in Angular+Typscript project?

**Comment by nbvishwa** [comment]
@slaven3kopic 

Hello plz find the below HTML & TS code. Is this the correct way of implementing it?

**Comment by slaven3kopic** [comment]
@nbvishwa Yes it is. It can be that you are not seeing 'https://www.google.com' because of override.

If this is the case try to erase the following code:

`
        this.webViewInterface.on(WebViewExt.shouldOverrideUrlLoadingEvent,
            (args2: ShouldOverideUrlLoadEventData) => {

            console.log('webview override');
            args2.cancel = true;

        });
`

**Comment by nbvishwa** [comment]
@slaven3kopic  Still facing the same issue.

**Comment by nbvishwa** [comment]
@slaven3kopic  can u show your complete ts code? Like is it a component or page?

**Comment by slaven3kopic** [comment]
@nbvishwa I am not allowed to do that. You can create a new issue to get more answers
